Build failure: premake5

## Steps To Reproduce

Steps to reproduce the behavior:

1. build premake

## Build log
```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/5qqnrc1928gk9w7f6ka6yfrd447k5wg9-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/ld2qca96a0v4v9rw4mpilnbwwqz3h32b-no-curl-ca.patch
patching file contrib/curl/premake5.lua
substituteStream() in derivation premake5-5.0.0-beta2: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'contrib/curl/premake5.lua')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
rm -rf ./bin
rm -rf ./build
rm -rf ./obj
mkdir -p build/bootstrap
gcc -o build/bootstrap/premake_bootstrap -DPREMAKE_NO_BUILTIN_SCRIPTS -DLUA_USE_POSIX -DLUA_USE_DLOPEN -I"contrib/lua/src" -I"contrib/luashim" src/host/*.c contrib/lua/src/lapi.c contrib/lua/src/lbaselib.c contrib/lua/src/lbitlib.c con>
./build/bootstrap/premake_bootstrap embed
Running action 'embed'...
Compiling... 
Embedding...
Writing...
Generated ../src/scripts.c...
./build/bootstrap/premake_bootstrap --to=build/bootstrap gmake2
Building configurations...
Running action 'gmake2'...
Generated build/bootstrap/Makefile...
Generated build/bootstrap/Premake5.make...
Generated build/bootstrap/lua-lib.make...
Generated build/bootstrap/luashim-lib.make...
Generated build/bootstrap/zlib-lib.make...
Generated build/bootstrap/zip-lib.make...
Generated build/bootstrap/mbedtls-lib.make...
Generated build/bootstrap/curl-lib.make...
Generated build/bootstrap/example.make...
Generated build/bootstrap/luasocket.make...
Done (734ms).
make -C build/bootstrap -j`getconf _NPROCESSORS_ONLN` config=release
make[1]: Entering directory '/build/source/build/bootstrap'
==== Building lua-lib (release) ====
==== Building zip-lib (release) ====
==== Building zlib-lib (release) ====
Creating obj/Release/lua-lib
==== Building curl-lib (release) ====
Creating bin/Release
==== Building mbedtls-lib (release) ====
==== Building luashim-lib (release) ====
lapi.c
Creating obj/Release/zlib-lib
lbaselib.c
adler32.c
compress.c
Creating obj/Release/luashim-lib
lbitlib.c
luashim.c
deflate.c
crc32.c
lcode.c
lcorolib.c
gzclose.c
lctype.c
gzlib.c
Creating obj/Release/zip-lib
mkstemp.c
Creating obj/Release/mbedtls-lib
aes.c
zip_add.c
aesni.c
Creating obj/Release/curl-lib
zip_add_dir.c
zip_add_entry.c
ldblib.c
amigaos.c
../../contrib/libzip/mkstemp.c: In function '_zip_mkstemp':
../../contrib/libzip/mkstemp.c:76:15: error: implicit declaration of function 'getpid' [-Wimplicit-function-declaration]
   76 |         pid = getpid();
      |               ^~~~~~
make[2]: *** [zip-lib.make:303: obj/Release/zip-lib/mkstemp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
asyn-ares.c
asyn-thread.c
arc4.c
make[1]: *** [Makefile:74: zip-lib] Error 2
make[1]: *** Waiting for unfinished jobs....
aria.c
axtls.c
base64.c
gzread.c
ldebug.c
cleartext.c
ldo.c
gzwrite.c
ldump.c
asn1parse.c
infback.c
inffast.c
inflate.c
Linking luashim-lib
conncache.c
inftrees.c
lfunc.c
asn1write.c
connect.c
lgc.c
base64.c
content_encoding.c
trees.c
bignum.c
cookie.c
uncompr.c
blowfish.c
linit.c
zutil.c
cram.c
liolib.c
camellia.c
ccm.c
curl_addrinfo.c
certs.c
llex.c
curl_des.c
curl_endian.c
curl_fnmatch.c
chacha20.c
lmathlib.c
curl_gethostname.c
chachapoly.c
cipher.c
lmem.c
curl_gssapi.c
cipher_wrap.c
loadlib.c
curl_memrchr.c
cmac.c
curl_multibyte.c
Linking zlib-lib
lobject.c
curl_ntlm_core.c
curl_ntlm_wb.c
ctr_drbg.c
lopcodes.c
debug.c
curl_rtmp.c
loslib.c
curl_sasl.c
curl_sspi.c
lparser.c
des.c
curl_threads.c
cyassl.c
darwinssl.c
lstate.c
dhm.c
dict.c
digest.c
ecdh.c
digest_sspi.c
lstring.c
dotdot.c
ecdsa.c
lstrlib.c
ecjpake.c
easy.c
ecp.c
ltable.c
escape.c
ltablib.c
file.c
fileinfo.c
ecp_curves.c
ltm.c
lundump.c
lutf8lib.c
lvm.c
lzio.c
formdata.c
entropy.c
entropy_poll.c
ftp.c
error.c
ftplistparser.c
getenv.c
gcm.c
getinfo.c
gopher.c
havege.c
hkdf.c
hmac_drbg.c
md.c
md2.c
gskit.c
gtls.c
md4.c
md5.c
memory_buffer_alloc.c
net_sockets.c
hash.c
nist_kw.c
hmac.c
hostasyn.c
oid.c
padlock.c
hostcheck.c
hostip.c
pem.c
hostip4.c
pk.c
hostip6.c
pk_wrap.c
hostsyn.c
pkcs11.c
pkcs12.c
pkcs5.c
http.c
pkparse.c
http2.c
pkwrite.c
http_chunks.c
http_digest.c
platform.c
http_negotiate.c
http_ntlm.c
http_proxy.c
platform_util.c
idn_win32.c
if2ip.c
poly1305.c
imap.c
inet_ntop.c
psa_crypto.c
psa_crypto_driver_wrappers.c
inet_pton.c
krb5.c
krb5_gssapi.c
psa_crypto_se.c
psa_crypto_slot_management.c
krb5_sspi.c
ldap.c
llist.c
psa_crypto_storage.c
mbedtls.c
md4.c
md5.c
memdebug.c
psa_its_file.c
mprintf.c
ripemd160.c
rsa.c
multi.c
netrc.c
rsa_internal.c
non-ascii.c
sha1.c
nonblock.c
nss.c
sha256.c
ntlm.c
ntlm_sspi.c
sha512.c
nwlib.c
ssl_cache.c
nwos.c
ssl_ciphersuites.c
oauth2.c
Linking lua-lib
ssl_cli.c
openldap.c
openssl.c
ssl_cookie.c
parsedate.c
pingpong.c
pipeline.c
ssl_msg.c
polarssl.c
polarssl_threadlock.c
pop3.c
progress.c
rand.c
rtsp.c
schannel.c
security.c
ssl_srv.c
ssl_ticket.c
select.c
ssl_tls.c
sendf.c
share.c
slist.c
ssl_tls13_keys.c
threading.c
timing.c
version.c
version_features.c
x509.c
x509_create.c
smb.c
smtp.c
socks.c
socks_gssapi.c
socks_sspi.c
speedcheck.c
splay.c
spnego_gssapi.c
x509_crl.c
x509_crt.c
spnego_sspi.c
ssh.c
x509_csr.c
strcase.c
x509write_crt.c
x509write_csr.c
xtea.c
strdup.c
strerror.c
strtok.c
strtoofft.c
system_win32.c
telnet.c
tftp.c
timeval.c
transfer.c
url.c
vauth.c
version.c
vtls.c
warnless.c
wildcard.c
x509asn1.c
Linking mbedtls-lib
Linking curl-lib
make[1]: Leaving directory '/build/source/build/bootstrap'
make: *** [Bootstrap.mak:113: linux] Error 2

```

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
```

</details>

## Additional context
This is the build tool of a package I maintain so I am affected

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 24.11 (Vicuna), 24.11.20241220.1c6e20d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos-24.11"`
 - nixpkgs: `/nix/store/b8l3ndracmly9kyylawrnk32lsrjfy3j-source`
```

## Notify maintainers
@bjornfor  

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/misc/premake/5.nix
+++ b/pkgs/development/tools/misc/premake/5.nix
@@ -10,13 +10,13 @@
 
 stdenv.mkDerivation rec {
   pname = "premake5";
-  version = "5.0.0-beta2";
+  version = "5.0.0-beta4";
 
   src = fetchFromGitHub {
     owner = "premake";
     repo = "premake-core";
     rev = "v${version}";
-    sha256 = "sha256-2R5gq4jaQsp8Ny1oGuIYkef0kn2UG9jMf20vq0714oY=";
+    sha256 = "sha256-sNLCyIHWDW/8jIrMFCZAqtWsh4SRugqtPR4HaoW/Vzk=";
   };
 
   buildInputs =

--- a/pkgs/development/tools/misc/premake/no-curl-ca.patch
+++ b/pkgs/development/tools/misc/premake/no-curl-ca.patch
@@ -11,25 +11,27 @@ diff --git a/contrib/curl/premake5.lua b/contrib/curl/premake5.lua
 index 474f5cfa..553bbd02 100644
 --- a/contrib/curl/premake5.lua
 +++ b/contrib/curl/premake5.lua
-@@ -32,19 +32,6 @@ project "curl-lib"
+@@ -36,21 +36,6 @@ project "curl-lib"
  
  		-- find the location of the ca bundle
  		local ca = nil
 -		for _, f in ipairs {
 -			"/etc/ssl/certs/ca-certificates.crt",
+-			"/etc/openssl/certs/ca-certificates.crt",
 -			"/etc/pki/tls/certs/ca-bundle.crt",
 -			"/usr/share/ssl/certs/ca-bundle.crt",
 -			"/usr/local/share/certs/ca-root.crt",
 -			"/usr/local/share/certs/ca-root-nss.crt",
 -			"/etc/certs/ca-certificates.crt",
--			"/etc/ssl/cert.pem" } do
+-			"/etc/ssl/cert.pem",
+-			"/boot/system/data/ssl/CARootCertificates.pem" } do
 -			if os.isfile(f) then
 -				ca = f
 -				break
 -			end
 -		end
  		if ca then
- 			defines { 'CURL_CA_BUNDLE="' .. ca .. '"' }
+ 			defines { 'CURL_CA_BUNDLE="' .. ca .. '"', 'CURL_CA_PATH="' .. path.getdirectory(ca) .. '"' }
  		end
 -- 
 2.37.2

